File: swift/701_InsertIntoABinarySearchTree.swift
class Solution {
    func insertIntoBST(_ node: TreeNode?, _ val: Int) -> TreeNode? {
        if node == nil { return TreeNode(val) }
        else if val < (node?.val)! { node?.left = insertIntoBST(node?.left, val) }
        else { node?.right = insertIntoBST(node?.right, val) }
        return node
    }
}

File: python/701_InsertIntoABinarySearchTree.py
class Solution:
    def insertIntoBST(self, root: TreeNode, val: int) -> TreeNode:
        # iterative approach
        node = root
        while node:
            if node.val > val:
                if not node.left:
                    node.left = TreeNode(val)
                    return root
                node = node.left
            else:
                if not node.right:
                    node.right = TreeNode(val)
                    return root
                node = node.right
        return TreeNode(val)
